src/ast/insert.rs

```diff
@@ -3,40 +3,82 @@ use crate::ast::*;
 /// A builder for an `INSERT` statement.
 #[derive(Clone, Debug, PartialEq)]
 pub struct Insert<'a> {
-    pub(crate) table: Table<'a>,
+    pub(crate) table: Option<Table<'a>>,
     pub(crate) columns: Vec<Column<'a>>,
-    pub(crate) values: Vec<Row<'a>>,
+    pub(crate) values: Expression<'a>,
     pub(crate) on_conflict: Option<OnConflict>,
     pub(crate) returning: Option<Vec<Column<'a>>>,
 }
 
 /// A builder for an `INSERT` statement for a single row.
 pub struct SingleRowInsert<'a> {
-    pub(crate) table: Table<'a>,
+    pub(crate) table: Option<Table<'a>>,
     pub(crate) columns: Vec<Column<'a>>,
     pub(crate) values: Row<'a>,
 }
 
 /// A builder for an `INSERT` statement for multiple rows.
 pub struct MultiRowInsert<'a> {
-    pub(crate) table: Table<'a>,
+    pub(crate) table: Option<Table<'a>>,
     pub(crate) columns: Vec<Column<'a>>,
     pub(crate) values: Vec<Row<'a>>,
 }
 
 #[derive(Clone, Copy, Debug, PartialEq)]
 /// `INSERT` conflict resolution strategies.
 pub enum OnConflict {
-    /// When a row already exists, do nothing.
+    /// When a row already exists, do nothing. Works with PostgreSQL, MySQL or
+    /// SQLite without schema information.
     ///
     /// ```rust
     /// # use quaint::{ast::*, visitor::{Visitor, Sqlite}};
+    /// # fn main() -> Result<(), quaint::error::Error> {
     /// let query: Insert = Insert::single_into("users").into();
     ///
-    /// let (sql, _) = Sqlite::build(query.on_conflict(OnConflict::DoNothing));
+    /// let (sql, _) = Sqlite::build(query.on_conflict(OnConflict::DoNothing))?;
     ///
     /// assert_eq!("INSERT OR IGNORE INTO `users` DEFAULT VALUES", sql);
+    /// # Ok(())
+    /// # }
     /// ```
+    ///
+    /// With Microsoft SQL server not supporting `IGNORE` in the `INSERT`
+    /// statement, the `INSERT` is converted to a `MERGE` statement. For it to work
+    /// in a correct way, the table should know all unique indices of the actual table.
```

Comment:
nitpick: it would be easier to understand by specifying a bit more, like "the quaint table should know all unique indices of the actual database table".